spec/generators/rspec/integration/integration_generator_spec.rb

```diff
@@ -4,5 +4,83 @@
 
 RSpec.describe Rspec::Generators::IntegrationGenerator, type: :generator do
   setup_default_destination
-  it_behaves_like "a request spec generator"
```

Comment:
Rather than removing this, it would be better to modify the shared example by passing in a flag to specifying when to define that extra spec.

Reply:
> So when the IntegrationGenerator class is deleted, should the shared examples stay there even though it will be only used one time?

No when they're deleted they can be pulled into the resulting specs, but thats not going to happen until the next major version.

> My interpretation was that the shared examples were only there because there was the inheritance of RequestGenerator < IntegrationGenerator. Adding a flag to an inheritance (which the shared examples are) is a code smell in my opinion.

No they are there because the two code paths, er, share examples, that there was an inheritance to share behaviour is an implementation detail, the examples would be valid if they both had vendored copies of the lass or completely different implementations.